chalk-ir/src/debug.rs

```diff
@@ -344,10 +344,10 @@ impl<T: Display> Display for Canonical<T> {
 }
 
 impl<T: Debug, L: Debug> Debug for ParameterKind<T, L> {
-    default fn fmt(&self, fmt: &mut Formatter) -> Result<(), Error> {
+    fn fmt(&self, fmt: &mut Formatter) -> Result<(), Error> {
         match *self {
-            ParameterKind::Ty(ref n) => write!(fmt, "Ty({:?})", n),
-            ParameterKind::Lifetime(ref n) => write!(fmt, "Lifetime({:?})", n),
```

Comment:
Hmm, this is annoying. Sometimes we use `ParameterKind<T, T>` for the same `T`, in which case it's pretty important to know whether it is `Ty` or `Lifetime`. OTOH yeah the output otherwise is kind of annoyingly verbose.

We could probably just make a copy of this type for the special case of `Parameter`, even if it means copying some methods or something. Not sure how much work that would be, but it might be worth it.